packages/vite/src/node/plugins/css.ts

```diff
@@ -1095,12 +1095,8 @@ function createCSSResolvers(config: ResolvedConfig): CSSAtImportResolvers {
           preferRelative: true,
         })
         sassResolve = async (...args) => {
-          const id = args[0]
-          if (id.startsWith('file://')) {
-            const fileUrl = new URL(id)
-            if (fs.existsSync(fileUrl)) {
-              return fileURLToPath(fileUrl)
-            }
```

Comment:
To explain what was happening, sass modern api calls a custom importer twice for each `@import` resolution. I made a simple test code here without Vite https://github.com/hi-ogawa/reproductions/tree/main/vite-17960-sass-legacy-directory-import

For the test case `@import "./dir"`, importer's `canonicalize` is called twice:

```js
// 1st call
url: 'file:///abs-path-to/scss-dir/dir',
importer: undefined

// 2nd call
url: 'dir'
importer: 'file://abs-path-to-/scss-dir/main.scss'
```

Before `file://` check is introduced in https://github.com/vitejs/vite/pull/17909, `@import "./dir"` was working due to 2nd call of `canonicalize`. After https://github.com/vitejs/vite/pull/17909, it's breaking since it's resolving to a directory.